core/src/main/java/io/kestra/core/runners/RunContext.java

```diff
@@ -461,6 +462,12 @@ public RunContext updateVariables(WorkerTaskResult workerTaskResult, TaskRun par
         return this.clone(variables);
     }
 
+    public RunContext addVariables(String nameProperty, Map<String, Object> variables) {
```

Comment:
Not a good idea.

You can render with an additional variable map if you need so better to not update the runcontext itself